ci/test.go

```diff
@@ -49,35 +63,62 @@ func (t *Test) Custom(
 	// +default="./..."
 	pkg string,
 	// +optional
+	failfast bool,
+	// +optional
+	parallel int,
+	// +optional
+	timeout string,
+	// +optional
 	race bool,
 ) error {
-	return t.test(ctx, race, run, pkg)
+	return t.test(ctx, run, pkg, failfast, parallel, timeout, race)
 }
 
 func (t *Test) test(
 	ctx context.Context,
-	race bool,
 	testRegex string,
 	pkg string,
+	failfast bool,
+	parallel int,
+	timeout string,
+	race bool,
 ) error {
 	cgoEnabledEnv := "0"
 	args := []string{
 		"gotestsum",
 		"--format", "testname",
 		"--no-color=false",
 		"--jsonfile=./tests.log",
-		"--",
-		// go test flags
-		"-parallel=16",
-		"-count=1",
-		"-timeout=30m",
 	}
 
+	if failfast {
+		args = append(args, "--max-fails=1")
+	}
+
+	// All following are go test flags
+	args = append(args, "--")
+
+	// Default parallel to number of CPUs
+	if parallel == 0 {
+		parallel = runtime.NumCPU()
+	}
+	args = append(args, fmt.Sprintf("-parallel=%d", parallel))
+
+	// Default timeout to 30m
+	// No test suite should take more than 30 minutes to run
+	if timeout == "" {
+		timeout = "30m"
+	}
+	args = append(args, fmt.Sprintf("-timeout=%s", timeout))
```

Comment:
We should use `// +default` for this one.

Reply:
If we did this, we would have the default in 3 places:
1. `All()`
2. `Important()`
3. `Custom()`

Since the `race` flag uses the same approach - default in code rather than `// +default` - I think that we should leave this as is. WDYT?

Reply:
Hm, alternatively, we should refactor all of these into one function - then which tests to run should probably be flags on in.

Not a blocker though, happy as is.